src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -26,8 +26,8 @@ private void resetSingleton() throws NoSuchFieldException, IllegalAccessExceptio
     public void setup() throws NoSuchFieldException, IllegalAccessException {
         usd = Currency.getInstance("USD");
         product = new Product("test-product", "Samsung Galaxy S", new BigDecimal(100), usd, 100, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg");
-        productDao = (ArrayListProductDao) ArrayListProductDao.getInstance();
         resetSingleton();
+        productDao = (ArrayListProductDao) ArrayListProductDao.getInstance();
```

Comment:
Here also would be cool if these two lines were in separate method, so as to avoid calling them separately. Anyway it's just a tip, no need to fix